### PR TITLE
Responsive content status in Content Selector #2887

### PR DESCRIPTION
**Pull request by ashklianko**
Requires https://github.com/enonic/lib-admin-ui/pull/1809

### CODE REVIEW

File: src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts [hunk 1]
```diff
@@ -67,10 +74,54 @@ export class ContentComboBox<ITEM_TYPE extends ContentTreeSelectorItem>
...
+            if (this.isStatusColumnShown()) {
+                const newColumns: GridColumn<any>[] = columns.filter((column: GridColumn<any>) => column.id !== 'status') ;
+                this.getDataGrid().setColumns(newColumns, true);
+            }
+        } else {
+            if (!this.isStatusColumnShown()) {
+                const newColumns: GridColumn<any>[] = [...columns, this.statusColumn];
+                this.getDataGrid().setColumns(newColumns, true);
+            }
+        }
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts]: I suggest we simplify this a bit.
1. Initially assign `newColumns = columns` (before `if...`)
2. Assign `this.isStatusColumnShown()` to a const (before `if...`). It's called either way, so we might as well just call it once.
3. After `if...` compare lengths of `columns` and `newColumns`. If they are different, call `this.getDataGrid().setColumns()`

File: src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts [hunk 2]
```diff
@@ -67,10 +74,54 @@ export class ContentComboBox<ITEM_TYPE extends ContentTreeSelectorItem>
...
+        ResponsiveManager.onAvailableSizeChanged(this, debouncedHandler);
+    }
+
+    private handleAvailableSizeChanged() {
+        const columns: GridColumn<any>[] = this.getColumnsWithoutCheckbox();
+        const width: number = this.getEl().getWidth();
+
+        if (ResponsiveRanges._360_540.isFitOrSmaller(width)) {
+            if (this.isStatusColumnShown()) {
+                const newColumns: GridColumn<any>[] = columns.filter((column: GridColumn<any>) => column.id !== 'status') ;
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts]: Compare with `this.statusColumn.getId()`.

File: src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts [hunk 3]
```diff
@@ -67,10 +74,54 @@ export class ContentComboBox<ITEM_TYPE extends ContentTreeSelectorItem>
...
+    private getDataGrid(): Grid<any> {
+        return this.getComboBox().getComboBoxDropdownGrid().getGrid();
+    }
+
+    private getColumnsWithoutCheckbox(): GridColumn<any>[] {
+        return this.getDataGrid().getColumns().filter((column: GridColumn<any>) => column.id !== '_checkbox_selector');
+    }
+
+    private isStatusColumnShown(): boolean {
+        return this.getColumnsWithoutCheckbox().some((column: GridColumn<any>) => column.id === 'status');
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts:127]: Compare with `this.statusColumn.getId()`.

**Review by alansemenov — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts (hunk 2 above)
Comment (ashklianko) [reply src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts]: Won't work: ssome of slickgrid objects (checkbox in our case) don't have `getId()` method

File: src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts (hunk 3 above)
Comment (ashklianko) [reply src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts:127]: Won't work: some of slickgrid objects (checkbox in our case) don't have `getId()` method , but have `id `property

File: src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts (hunk 1 above)
Comment (ashklianko) [reply src/main/resources/assets/js/app/inputtype/ui/selector/ContentComboBox.ts]: Reworked in other way : extracted to small functions with self describing names

**Review by alansemenov — APPROVED** [review]

**Pull request merged**